hashing.py

```
@@ -0,0 +1,70 @@
+    logger.debug(f"tokens: {tokens}")
+    n = len(tokens)
+    logger.debug(f"n: {n}")
+    for i in range(n):
+        for j in range(i+1, n+1):
+            logger.debug(f"i: {i}, j: {j}")
+            sub_query = " ".join(tokens[i:j])
+            logger.debug(f"sub_query: {sub_query}")
+            if not sub_query[-1].isalnum():
+                sub_query = sub_query[:-1]
+            logger.debug(f"sub_query: {sub_query}")
+            df = con.sql(f"select hash from {TABLE_NAMES['MG_HASH_MAP']} where mg_id = '{sub_query}'").to_df()
```

Ruff (0.8.2)

Possible SQL injection vector through string-based query construction

(S608)